awx/main/scheduler/dag_simple.py

```diff
@@ -28,97 +60,166 @@ def __iter__(self):
         return self.nodes.__iter__()
 
     def generate_graphviz_plot(self):
-        def short_string_obj(obj):
-            if type(obj) == Job:
-                type_str = "Job"
-            if type(obj) == AdHocCommand:
-                type_str = "AdHocCommand"
-            elif type(obj) == InventoryUpdate:
-                type_str = "Inventory"
-            elif type(obj) == ProjectUpdate:
-                type_str = "Project"
-            elif type(obj) == WorkflowJob:
-                type_str = "Workflow"
-            else:
-                type_str = "Unknown"
-            type_str += "%s" % str(obj.id)
-            return type_str
+        def run_status(obj):
```

Comment:
if `obj` is a Django `WorkflowJobNode` object, then why not just have this a method on the model?